### PR TITLE
Test rule

### PR DESCRIPTION
**Pull request by annethyme**
<< Describe the changes >>

Closes issue: none

## Guidance for the PR (pull request) creator

**When creating PR:**
- [ ] Make sure you requesting to **pull a issue/feature/bugfix branch** (right side) to the master branch (left side).

**After creating PR:**
- [x] Add yourself (and co-authors) as "Assignees" for PR
- [x] Add label to indicate if it's a `Rule`, `Definition` or `Chore` (more to the administrative side)
- [x] Add relevant project (e.g. "Q3 2018 Status") to PR
- [x] OPTIONAL: If you want anyone in particular to review your pull request, assign them as "Reviewers".
- [ ] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)

# How to Review And Approve
- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.

### CODE REVIEW

File: _rules/test.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+name: Test
+description: |
+  Each form field element has an accessible name
+
+success_criterion:
+- 3.3.2 # Labels or Instructions
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
```
Comment (jwjhix) [inline _rules/test.md]: TEst comment on Dom tree

File: _rules/test.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+## Test procedure
+
+### Applicability
+
+This rule applies to any element that is [included in the accessibility tree](#included-in-the-accessibility-tree), or [focusable](#focusable) and that has one of the following [semantic roles](#semantic-role): `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider, spinbutton`, `switch`, `textbox`.
+
+**Note**: The list of roles is derived by taking all the [ARIA](https://www.w3.org/TR/wai-aria-1.1/) 1.1 roles that:
+- have a [semantic roles](#semantic-role) that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
+- does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+- The `option` role is not part of the list of applicable roles, because it does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
```
Comment (jwjhix) [inline _rules/test.md]: Another comment on the file

File: _rules/test.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  <div role="textbox" data-rule-target></div>
+</label>
+```
+
+#### Failed example 6
+
+Explicit label not supported on div elements.
+
+```html
+<label for="lastname">first name</label>
```
Comment (jwjhix) [inline _rules/test.md]: Pas possible à l'instant

File: _rules/test.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
+---
+name: Test
+description: |
+  Each form field element has an accessible name
```
Comment (jorgeponto) [inline _rules/test.md]: aaaa aa

File: _rules/test.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+- have a [semantic roles](#semantic-role) that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
+- does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+- The `option` role is not part of the list of applicable roles, because it does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is [non-empty](#non-empty).
+
+## Assumptions
+
```
Comment (kengdoj) [inline _rules/test.md]: is this a good place to address disabled and read-only form fields?

File: _rules/test.md (hunk 1 above)
Comment (cmccardoso) [reply _rules/test.md]: really? I don't think so

File: _rules/test.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+  Each form field element has an accessible name
+
+success_criterion:
+- 3.3.2 # Labels or Instructions
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+
+authors:
```
Comment (john-urbilog) [inline _rules/test.md]: test test

**Pull request closed without merging**